src/activities/Elsa.Activities.MassTransit/Activities/PublishMassTransitMessage.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Elsa.Attributes;
+using Elsa.Expressions;
+using Elsa.Results;
+using Elsa.Services;
+using Elsa.Services.Models;
+using MassTransit;
+
+namespace Elsa.Activities.MassTransit.Activities
+{
+    [ActivityDefinition(
+        Category = "MassTransit",
+        DisplayName = "Publish MassTransit Message",
+        Description = "Publish an event via MassTransit."
+    )]
+    public class PublishMassTransitMessage : Activity
+    {
+        private readonly ConsumeContext consumeContext;
+        private readonly IBus bus;
+        private readonly IWorkflowExpressionEvaluator evaluator;
+
+        public PublishMassTransitMessage(ConsumeContext consumeContext, IBus bus, IWorkflowExpressionEvaluator evaluator)
```

Comment:
Just being curious: how does the service provider know what `ConsumeContext` to inject?

Reply:
After this comment I now have reservations that this might not work with all container implementations, so I might test a little further.

Basically MassTransit registers `ConsumeContext` interface as scoped. So when you are within the scope of a consumer, the container will contain the `ConsumeContext instance.

Here I have modified Sample08 to publish a message before the timer starts. You can see that I have a handle to the consume context, and the callstack starts from the consume of the `CreateOrder` command.
![image](https://user-images.githubusercontent.com/164331/69284486-a68d3880-0c3e-11ea-8adb-bd434b247530.png)

However after the timer, there is no consume context within the current scope to publish the `OrderShipped` event.
![image](https://user-images.githubusercontent.com/164331/69284693-1d2a3600-0c3f-11ea-864b-36e445b8322f.png)

So this event will be published using the bus.

Reply:
Well if Unity container works, then I would assume that all others should also work.

![image](https://user-images.githubusercontent.com/164331/69285577-23211680-0c41-11ea-88e7-b6d0694695a0.png)

Reply:
Great answer. I didn’t know this about MT.